### PR TITLE
Remove inbox feature flag

### PR DESCRIPTION
**Pull request by szabinst**
refs: MBL-17721
affects: Student, Teacher
release note: Inbox has been redesigned.
test plan: Test inbox functionality

## Checklist

- [ ] Follow-up e2e test ticket created
- [ ] A11y checked
- [ ] Tested on phone
- [x] Tested on tablet
- [x] Tested in dark mode
- [ ] Tested in light mode
- [x] Approve from product

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
#### Release Note: 
Inbox has been redesigned.

#### Affected Apps: Student, Teacher

[MBL-17721](https://instructure.atlassian.net/browse/MBL-17721)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 91.59% | 91.5% | 0.1%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2222b5040722e41d3b4520287100390fc329981c
</p>

**Review by balintbartok — DISMISSED** [review]
QA + 1, opened a followup ticket about unarchive
https://instructure.atlassian.net/browse/MBL-17759

File: Core/Core/Inbox/View/InboxView.swift [hunk 1]
```diff
@@ -115,6 +119,27 @@ public struct InboxView: View, ScreenViewTrackable {
...
+                } icon: {
+                    Image.unarchiveLine
+                        .foregroundColor(.textLightest)
+                }
+                .labelStyle(.iconOnly)
+            }
+            .tint(.ash)
+        } else {
+            SwiftUI.EmptyView()
+        }
```
Comment (rh12) [inline Core/Core/Inbox/View/InboxView.swift:140]: nit: Do we need the `else` in this case? Won't the `@ViewBuilder` do the same if we omit it?

**Review by rh12 — APPROVED** [review]
(One minor question, no need to change it unless you push something anyway)

File: Core/Core/Inbox/View/InboxView.swift (hunk 1 above)
Comment (szabinst) [reply Core/Core/Inbox/View/InboxView.swift:140]: I'm not sure, this code is copy-paste from the already existing archive button.

File: Core/Core/Inbox/MessageDetails/MessageDetailsAssembly.swift [hunk 2]
```diff
@@ -20,10 +20,18 @@ import Foundation
 
 public enum MessageDetailsAssembly {
 
-    public static func makeViewController(env: AppEnvironment,
-                                          conversationID: String) -> UIViewController {
+    public static func makeViewController(
+        env: AppEnvironment,
+        conversationID: String,
+        allowArchive: Bool
```
Comment (vargaat) [inline Core/Core/Inbox/MessageDetails/MessageDetailsAssembly.swift:26]: We should check if we can decide the value of this property inside the assembly or in the view model / interactor. The reason is that messages are frequently opened from push notifications where this url parameter won't be available.

File: Core/CoreTests/Extensions/InstIconExtensionsTests.swift [hunk 3]
```diff
@@ -459,60 +463,6 @@ class InstIconExtensionTests: XCTestCase {
...
-        XCTAssertEqual(Image.masteryLTI, Image("masteryLTI", bundle: .core))
-        XCTAssertEqual(Image.offlineLine, Image("offlineLine", bundle: .core))
-        XCTAssertEqual(Image.partialSolid, Image("partialSolid", bundle: .core))
-        XCTAssertEqual(Image.radioButtonSelected, Image("radioButtonSelected", bundle: .core))
-        XCTAssertEqual(Image.radioButtonUnselected, Image("radioButtonUnselected", bundle: .core))
-        XCTAssertEqual(Image.qrCode, Image("qrCode", bundle: .core))
-        XCTAssertEqual(Image.share, Image("share", bundle: .core))
-        XCTAssertEqual(Image.todoTab, Image("todoTab", bundle: .core))
-        XCTAssertEqual(Image.todoTabActive, Image("todoTabActive", bundle: .core))
-        XCTAssertEqual(Image.unionLine, Image("unionLine", bundle: .core))
```
Comment (vargaat) [inline Core/CoreTests/Extensions/InstIconExtensionsTests.swift:515]: Let's either move this to another file or discard this change for now.

**Review by vargaat — CHANGES_REQUESTED** [review]

File: Core/Core/Inbox/MessageDetails/MessageDetailsAssembly.swift (hunk 2 above)
Comment (szabinst) [reply Core/Core/Inbox/MessageDetails/MessageDetailsAssembly.swift:26]: We don't have that information, hence this solution.

File: Core/CoreTests/Extensions/InstIconExtensionsTests.swift (hunk 3 above)
Comment (szabinst) [reply Core/CoreTests/Extensions/InstIconExtensionsTests.swift:515]: Added to a new file.

**Review by vargaat — APPROVED** [review]

**Review by ndrsszsz — APPROVED** [review]
QA +1
Tested on iPad, iOS 16.7.1

**Pull request merged**